File: Advanced-Business-Development-With -DotNET/Models/Candidatura.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobFitScoreAPI.Models
{
    [Table("CANDIDATURAS")]
    public class Candidatura
    {
        [Key]
        [Column("id_candidatura")]
        public int IdCandidatura { get; set; }

        [Required]
        [Column("usuario_id")]
        public int UsuarioId { get; set; }

        [Required]
        [Column("vaga_id")]
        public int VagaId { get; set; }

        [Column("data_candidatura")]
        public DateTime DataCandidatura { get; set; } = DateTime.Now; 

        [Column("status")]
        [MaxLength(50)]
        public string Status { get; set; } = "Em Análise"; 

        
        [ForeignKey(nameof(UsuarioId))]
        public Usuario? Usuario { get; set; }

        [ForeignKey(nameof(VagaId))]
        public Vaga? Vaga { get; set; }
    }
}

File: Java-Advanced/src/main/java/com/gs/fiap/jobfitscore/domain/candidatura/Candidatura.java
package com.gs.fiap.jobfitscore.domain.candidatura;

import com.gs.fiap.jobfitscore.domain.usuario.Usuario;
import com.gs.fiap.jobfitscore.domain.vaga.Vaga;
import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "candidaturas")
public class Candidatura {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long idCandidatura;
	
	@ManyToOne(optional = false)
	@JoinColumn(name = "usuario_id", nullable = false)
	private Usuario usuario;
	
	@ManyToOne(optional = false)
	@JoinColumn(name = "vaga_id", nullable = false)
	private Vaga vaga;
	
	@Column(name = "data_candidatura", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
	private LocalDateTime dataCandidatura = LocalDateTime.now();
	
	@Column(length = 50)
	private String status = "Em Análise";
	
	public Long getIdCandidatura() {
		return idCandidatura;
	}
	
	public void setIdCandidatura( Long idCandidatura ) {
		this.idCandidatura = idCandidatura;
	}
	
	public Usuario getUsuario() {
		return usuario;
	}
	
	public void setUsuario( Usuario usuario ) {
		this.usuario = usuario;
	}
	
	public Vaga getVaga() {
		return vaga;
	}
	
	public void setVaga( Vaga vaga ) {
		this.vaga = vaga;
	}
	
	public LocalDateTime getDataCandidatura() {
		return dataCandidatura;
	}
	
	public void setDataCandidatura( LocalDateTime dataCandidatura ) {
		this.dataCandidatura = dataCandidatura;
	}
	
	public String getStatus() {
		return status;
	}
	
	public void setStatus( String status ) {
		this.status = status;
	}
}
